src/BaGet.Core/Decompiler/SourceCode/SourceCodeDecompiler.cs

```diff
@@ -0,0 +1,39 @@
+using System;
+using BaGet.Core.Entities;
+using ICSharpCode.Decompiler.CSharp;
+using ICSharpCode.Decompiler.TypeSystem;
+
+namespace BaGet.Core.Decompiler.SourceCode
+{
+    internal class SourceCodeDecompiler : ISourceCodeProvider
+    {
+        private readonly CSharpDecompiler _decompiler;
+
+        public SourceCodeDecompiler(CSharpDecompiler decompiler)
+        {
+            _decompiler = decompiler;
```

Comment:
Please add null checks to the constructor: `_decompiler = decompiler ?? throw new ArgumentNullException(nameof(decompiler));`